lib/googleauth/compute_engine.rb

```diff
@@ -85,7 +85,8 @@ def fetch_access_token options = {}
         c = options[:connection] || Faraday.default_connection
         retry_with_error do
           uri = target_audience ? COMPUTE_ID_TOKEN_URI : COMPUTE_AUTH_TOKEN_URI
-          query = target_audience ? { "audience" => target_audience, "format" => "full" } : nil
+          query = target_audience ? { "audience" => target_audience, "format" => "full" } : {}
+          query[:scopes] = scope.join " " if scope
```

Comment:
According to the signet docs, `scope` could either be an array of strings, or a space-delimited string. Need to handle both cases. Maybe:
```ruby
query[:scopes] = Array(scope).join " " if scope
```

Reply:
Thanks @dazuma.
Rebased and addressed.